### ISSUE TITLE
Travis dies from vet and lint errors since #207 was merged

### ISSUE THREAD

**Issue by jackgr** [kubernetes/deployment-manager#223]
`The command "make setup-gotools test" exited with 2.`

Automated builds are broken until this issue is fixed.

/cc @michelleN, @sgoings

**Comment by sgoings** [comment]
> Automated builds are broken until this issue is fixed.

Isn't _quite_ a correct statement. The reality is that current master is stable and still happily building in Travis. This build: https://travis-ci.org/kubernetes/deployment-manager/builds/107148732 is the build on master with @michelleN's change merged into master.

The issue I think you're referring to is that certain PR builds are failing due to the fact that the changes within PRs are cause some complaint from go vet/lint. An example would be this PR build: https://travis-ci.org/kubernetes/deployment-manager/builds/107360379#L408-L410

On lines 408 - 410 go vet is complaining about:

```
repository/persistent/persistent.go:35: struct field tag "_id" not compatible with reflect.StructTag.Get: bad syntax for struct tag pair
```

It's possible to teach vet to be less strict on some things... however I personally feel like it'd probably be best to stick with golang's vet suggestions.

**Comment by jackgr** [comment]
@sgoings yes, I realized that. Fixed all the vet and lint errors in master. Then rebased persistence.

**Comment by sgoings** [comment]
Just following up...

`go lint` is a tool that suggests changes but doesn't return non-zero exit codes when it finds things.
`go vet` is a tool that enforces change. It returns non-zero exit codes when it finds things.

`go lint` was producing lots of output + suggestions which you fixed up via #224 (Thanks! :+1:) but wasn't actually hindering the build process... just maybe being a bit too chatty :smile:. The `go vet` errors are still in existence with PR #224. `go vet` isn't happy in particular with:

persistence.go lines: [35](https://github.com/kubernetes/deployment-manager/pull/170/files#diff-18acdc1aadcd8a42d3ce6fdf6f7d80c6R35) [40](https://github.com/kubernetes/deployment-manager/pull/170/files#diff-18acdc1aadcd8a42d3ce6fdf6f7d80c6R40) [45](https://github.com/kubernetes/deployment-manager/pull/170/files#diff-18acdc1aadcd8a42d3ce6fdf6f7d80c6R45)

Which are changes in a new file introduced in that PR.

**Comment by jackgr** [comment]
Yes, thanks.

Jack Greenfield | Google
